src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Javac.targets

```diff
@@ -0,0 +1,140 @@
+      JavaLibraries="@(AndroidJavaLibrary)"
+      ExternalJavaLibraries="@(AndroidExternalJavaLibrary)"
+      LibraryProjectJars="@(ExtractedJarImports)"
+      DoNotPackageJavaLibraries="@(_ResolvedDoNotPackageAttributes)"
+      EnableInstantRun="$(_InstantRunEnabled)"
+      >
+      <Output TaskParameter="JavaLibrariesToCompile" ItemName="_JavaLibrariesToCompile" />
+      <Output TaskParameter="ReferenceJavaLibraries" ItemName="_ReferenceJavaLibs" />
+    </DetermineJavaLibrariesToCompile>
+
+    <ItemGroup>
+      <_JavaLibrariesToCompileForApp Include="@(_JavaLibrariesToCompile)" Condition=" '$(_InstantRunEnabled)' != 'True' " />
+      <AndroidExternalJavaLibrary    Include="@(_JavaLibrariesToCompile)" Condition=" '$(_InstantRunEnabled)' == 'True' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CollectJavaSourceForBinding">
+    <ItemGroup>
+      <_JavaBindingSource Include="%(AndroidJavaSource.Identity)" Condition=" '%(AndroidJavaSource.Bind)' == 'True' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CollectJavaSource">
+    <ItemGroup>
+      <_JavaSource Include="%(AndroidJavaSource.Identity)" Condition=" '%(AndroidJavaSource.Bind)' != 'True' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CompileBindingJava"
+      Condition=" '@(_JavaBindingSource->Count())' != '0' "
+      DependsOnTargets="$(_CompileBindingJavaDependsOnTargets)"
+      Inputs="@(_AndroidMSBuildAllProjects);$(MonoPlatformJarPath);@(_JavaBindingSource)"
+      Outputs="$(_AndroidIntermediateBindingClassesZip)">
+
+    <!-- remove existing <Javac /> outputs, since *.class files and classes.zip could contain old files -->
+    <RemoveDirFixed Directories="$(_AndroidIntermediateBindingJavaClassDirectory)" Condition="Exists('$(_AndroidIntermediateBindingJavaClassDirectory)')" />
+    <Delete Files="$(_AndroidIntermediateBindingClassesZip)" Condition="Exists('$(_AndroidIntermediateBindingClassesZip)')" />
+    <!-- Compile java code -->
+    <Javac
+      Condition=" '@(_JavaBindingSource->Count())' != '0' "
+      JavaPlatformJarPath="$(JavaPlatformJarPath)"
+      ClassesOutputDirectory="$(_AndroidIntermediateBindingJavaClassDirectory)"
+      ClassesZip="$(_AndroidIntermediateBindingClassesZip)"
+      StubSourceDirectory="$(_AndroidIntermediateBindingJavaSourceDirectory)"
+      JavaSourceFiles="@(_JavaBindingSource)"
+      ToolPath="$(JavacToolPath)"
+      ToolExe="$(JavacToolExe)"
+      Jars="@(_BindingJavaLibrariesToCompile);@(_ReferenceJavaLibs)"
+      JavacTargetVersion="$(JavacTargetVersion)"
+      JavacSourceVersion="$(JavacSourceVersion)"
+    />
+    <ItemGroup>
+      <EmbeddedJar Include="$(_AndroidIntermediateBindingClassesZip)" Condition="Exists('$(_AndroidIntermediateBindingClassesZip)')" />
+      <!-- <AndroidJavaLibrary Include="$(_AndroidIntermediateBindingClassesZip)" Pack="True" Bind="True" Condition="Exists('$(_AndroidIntermediateBindingClassesZip)') And '$(UsingAndroidNETSdk)' == 'True'" /> -->
```

Comment:
I think this would be `@(EmbeddedJar)` if `%(Bind)` is `true` and `@(AndroidJavaLibrary)` otherwise.

You might add a test for both cases.

Reply:
Well if there are no `AndroidJavaSource` items with `Bind` is `true` , we won't even build  the `$(_AndroidIntermediateBindingClassesZip)`. So do we need this additional `AndroidJavaLibrary` bits? I don't think so.

Reply:
Ok, that brings up a question, if I look at:

https://github.com/xamarin/xamarin-android/blob/13ea326da0459d7d24d9b265796a21b8a2516bfa/src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.AvailableItems.targets#L90-L91

Can you use `@(AndroidJavaSource)` in a class library and *not* bind it? You would want the `.jar` packed up as a `@(AndroidJavaLibrary)` and no C# bindings.